lms/djangoapps/courseware/middleware.py

```diff
@@ -22,3 +25,21 @@ def process_exception(self, _request, exception):
                     args=[course_key.to_deprecated_string()]
                 )
             )
+
+
+class RedirectExpiredCoursesMiddleware(object):
```

Comment:
Rename to `ThemeRedirectMiddleware` and move to `openedx.core.djangoapps.theming.middleware.py`.